src/edge_kernels/StreletsUpwindEdgeAlg.C

```diff
@@ -0,0 +1,204 @@
+// Copyright 2017 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS), National Renewable Energy Laboratory, University of Texas Austin,
+// Northwest Research Associates. Under the terms of Contract DE-NA0003525
+// with NTESS, the U.S. Government retains certain rights in this software.
+//
+// This software is released under the BSD 3-clause license. See LICENSE file
+// for more details.
+//
+
+#include <edge_kernels/StreletsUpwindEdgeAlg.h>
+#include <Realm.h>
+#include "stk_mesh/base/NgpField.hpp"
+#include <ngp_utils/NgpFieldUtils.h>
+#include <ngp_utils/NgpLoopUtils.h>
+#include <SolutionOptions.h>
+
+namespace sierra {
+namespace nalu {
+
+StreletsUpwindEdgeAlg::StreletsUpwindEdgeAlg(
+  Realm& realm, stk::mesh::Part* part)
+  : Algorithm(realm, part),
+    velocityName_("velocity"),
+    pecletFactor_(get_field_ordinal(
+      realm_.meta_data(), "peclet_factor", stk::topology::EDGE_RANK)),
+    fOne_(get_field_ordinal(realm.meta_data(), "sst_f_one_blending")),
+    dualNodalVolume_(get_field_ordinal(realm.meta_data(), "dual_nodal_volume")),
+    sstMaxLen_(get_field_ordinal(realm.meta_data(), "sst_max_length_scale")),
+    dudx_(get_field_ordinal(realm.meta_data(), "dudx")),
+    density_(get_field_ordinal(realm.meta_data(), "density")),
+    viscosity_(get_field_ordinal(realm.meta_data(), "viscosity")),
+    turbViscosity_(get_field_ordinal(realm.meta_data(), "turbulent_viscosity")),
+    turbKE_(get_field_ordinal(realm.meta_data(), "turbulent_ke")),
+    specDissRate_(
+      get_field_ordinal(realm.meta_data(), "specific_dissipation_rate")),
+    edgeAreaVec_(get_field_ordinal(
+      realm.meta_data(), "edge_area_vector", stk::topology::EDGE_RANK)),
+    coordinates_(get_field_ordinal(
+      realm.meta_data(), realm.solutionOptions_->get_coordinates_name())),
+    velocity_(get_field_ordinal(realm.meta_data(), velocityName_))
+{
+  // TODO(psakiev) Double check upwind quantities have been populated when this
+  // gets constructed
+  const DblType alpha = realm_.get_alpha_factor(velocityName_);
+  const DblType alphaUpw = realm_.get_alpha_upw_factor(velocityName_);
+  const DblType hoUpwind = realm_.get_upw_factor(velocityName_);
+  // check that upwinding factors are set so we only blend with the peclet
```

Comment:
@rcknaus that was the intent here.  Are you thinking we check and throw in solution options or something? I was also thinking we could add strelets upwinding as an option within solution options at some point so it could be used with sst_des as well

Reply:
Yeah, more or less.  Like having strelets on automatically switches the blending coefficients to the values they have to be, and we check that the user isn't setting them to something else in solution options.

Reply:
@rcknaus I started looking into this, but it wasn't quite as straight forward as I thought to detect that upwind parameters were being parsed/set after checking the turbulence model.  Probably not hard to implement, but I think the version I have now is adequate for the intended purposes.  

Also, after reviewing the upwinding again I'm not really sure we need to error out over these parameters since @gantech's original implementation included the limiters in the upwind calculation.  The upwinding parameters are pretty much what we usually run anyways. Personally, I'd rather change this to a warning instead of an error.